### PR TITLE
Added Docker Image worflow and update other workflow

### PR DESCRIPTION
**Pull request by Gauvino**
## Checklist

- [X] I have read the [Contributing Guidelines]
- [X] I acknowledge that any submitted code will be licensed under the [ISC License]
- [X] I confirm that submitted code is my own work
- [X] I have tested the code, and confirm that it works

## Enviroment

- Operating System: null
- Node version: 20.8.0
- npm version: 10.2.0

## Description

Added Github workflow to build a Docker Image and automatically push it on Docker Hub
Upgraded the Github action version and in prevision make everything working for the new LTS of NodeJS 20

The Github workflow ASS CI build and test the build ASS then makes a Docker Image of it, and after this push it to Docker Hub, when all of the previous stuff is done, it can send Discord notification everything is ready or not.

This would help a lot of people on deploying ASS on Portainer instance or just with Docker Compose, automated everything and maybe if you want it, you can add cypress or eslint or both to the repository so I can make a workflow for testing code.

[Contributing Guidelines]: https://github.com/tycrek/ass/blob/master/CONTRIBUTING.md
[ISC License]: https://github.com/tycrek/ass/blob/master/LICENSE

### CODE REVIEW

File: .github/workflows/ts-build.yml
Comment (tycrek) [inline .github/workflows/ts-build.yml:1]: All looks good here :)

File: .github/workflows/docker-build [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tycrek) [inline .github/workflows/docker-build:48]: I'm opting to not use GHCR so remove these lines

File: .github/workflows/docker-build
Comment (tycrek) [inline .github/workflows/docker-build:1]: Overall looks good, just one comment within

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -32,20 +32,21 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        language: [ 'javascript' ]
+        language: [ 'javascript-typescript' ]
```
Comment (tycrek) [inline .github/workflows/codeql-analysis.yml:35]: [Future reference](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning) (ignore this comment)

**Review by tycrek — CHANGES_REQUESTED** [review]
Just the one thing to change (see review comments)

**Comment by Gauvino** [comment]
LGTM !

**Review by tycrek — APPROVED** [review]

**Comment by tycrek** [comment]
All looks good! Just going to set up Secrets stuff prior to merge

File: .github/workflows/docker-build.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+name: "Docker Build"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (tycrek) [inline .github/workflows/docker-build.yml:7]: One last thing, can you add `dev/0.15.0` branch to these arrays (**not** push? That way we can test it live (don't worry about the `tags: | \n tycrek/ass:master` below, that can stay as-is).

**Review by tycrek — CHANGES_REQUESTED** [review]
One tiny thing

**Review by tycrek — APPROVED** [review]
image

**Pull request merged**